### DISCUSSION TITLE
Making browser download a file written by python pyodide

### DISCUSSION THREAD

**Discussion by SKbarbon** [pyodide/pyodide#3885] [Q&A]
How do I make the browser download a file that written using python of pyodide ?

I really searched on the internet, but all of them not worked

**Comment by ryanking13** [comment]
You can read the file inside Pyodide filesystem from JavaScript using [`pyodide.FS.readFile`](https://emscripten.org/docs/api_reference/Filesystem-API.html#FS.readFile), then you can use other tools like [FileSaver.js](https://github.com/eligrey/FileSaver.js).

**Comment by JeffersGlass** [comment]
For a minimal use case, you might try [this recipe for File Download](https://pyscript.recipes/basic/file-download/). It's not quite as feature-ful as something like FileSaver.js, but may be simpler to drop into existing Python code.
